packages/expo-updates/android/src/main/java/expo/modules/updates/manifest/NewManifest.java

```diff
@@ -36,28 +35,30 @@ private NewManifest(JSONObject manifestJson,
                       String scopeKey,
                       Date commitTime,
                       String runtimeVersion,
-                      JSONObject metadata,
-                      Uri bundleUrl,
+                      JSONObject launchAsset,
                       JSONArray assets) {
     mManifestJson = manifestJson;
     mId = id;
     mScopeKey = scopeKey;
     mCommitTime = commitTime;
     mRuntimeVersion = runtimeVersion;
-    mMetadata = metadata;
-    mBundleUrl = bundleUrl;
+    mLaunchAsset = launchAsset;
     mAssets = assets;
   }
 
-  public static NewManifest fromManifestJson(JSONObject manifestJson, UpdatesConfiguration configuration) throws JSONException {
+  public static NewManifest fromManifestJson(JSONObject rootManifestJson, UpdatesConfiguration configuration) throws JSONException {
```

Comment:
This seems like it may miss the the manifest.

The new manifest is being served as an `apiv2` endpoint. This means that it is nested under a top level key `data`.

So instead of:
```
{
  manifest: {
    id: ....
  }
}
```
we will have:

```
{
  data :{
    publicManifest: {
      manifest: {
        id: ....
      }
  }
}
```

for reference: https://staging.exp.host/--/api/v2/update-manifest-experimental/9e3ebb0e-58f4-4bcf-854a-a3f74fb8d79e?native-runtime-version=1&channel-name=demo&platform=android

Reply:
Also would be really excited if you have a better name for `publicManifest`.

Reply:
Ok, I've amended the branch to support this format. A couple questions -- 
1) I think we probably want to be a bit lenient and not expect manifests from all servers to have these top-level `data` and `publicManifest` fields, and handle manifests that just start at the root level as well. Does that seem right ot you?
2) What's the purpose of the `publicManifest` key? To me it seems like an unnecessary second level of nesting but I definitely might be missing something.
3) How will the structure differ when we start providing signed manifests -- or is that not something we're including in the first milestone for EAS update?

Reply:
Great questions.

1. That sounds very right.
2. We chose `{ data: { publicManifest }}` instead of `{data: { ...publicManifest}}` as that would allow us flexibility in adding top level fields in the future. We don't have anything in mind, but it seemed a low cost way to prevent a potentially costly refactor.
3. I've left the signing out for the first milestone. But one idea (now that you mention it) is that we could put the signature as another top level field: `{ data:{ publicManifest, signature }}`

Reply:
Gotcha -- 1 and 2 make sense. For 3, if that's not included in the first milestone I think I will hold off on building support for that exact structure into the clients as it would require a little bit more restructuring, if that sounds ok to you.

Reply:
That sounds great to me.

Reply:
Hey, just got done talking with james, and we are actually going to try and move this back off the apiv2 endpoint.

It seems like the code you wrote doesn't care about that, so thank you for having this great foresight!

I'm going to try to get the PR for moving this endpoint landed ASAP, so it may be useful to hold off landing this PR till then (unless there is urgency or some other deadline, then please go ahead!).

Reply:
(sorry for going back and forth here)